### PR TITLE
Add support for admin registration to waiting list

### LINKED ISSUE #890: Admin registration to waiting list
I think we should support admin register to waiting list

### PR DESCRIPTION
**Pull request by edvardvb**
Adds support for admin registrations to the waiting list by allowing `pool=None` in the admin registration method, and making the pool not required in the serializer.

Closes #890

### CODE REVIEW

File: lego/apps/events/models.py [hunk 1]
```diff
@@ -84,6 +84,20 @@ def admin_register(self, user, pool, admin_reason, feedback=''):
         :param feedback: Feedback to organizers
         :return: The registration
         """
+        if not pool:
```
Comment (hanswilw) [inline lego/apps/events/models.py]: Can you write this without the duplicate transaction and update_or_create? `if not pool or self.pools.filter(id=pool.id):` and handle the locked pool counter in another if pool statement?

File: lego/apps/events/tests/test_admin_registrations.py [hunk 2]
```diff
@@ -122,6 +122,17 @@ def test_ar_twice(self):
...
         e1_no_of_regs_before = event.number_of_registrations
 
-        event.admin_register(user, pool, admin_reason='test')
-        event.admin_register(user, pool, admin_reason='test')
+        event.admin_register(user, admin_reason='test', pool=pool)
+        event.admin_register(user, admin_reason='test', pool=pool)
         self.assertEqual(event.number_of_registrations, e1_no_of_regs_before+1)
+
+    def test_ar_without_pool(self):
+        """Test that user is not registered twice when admin registered is run twice"""
```
Comment (hanswilw) [inline lego/apps/events/tests/test_admin_registrations.py]: Description needs update

File: lego/apps/events/models.py [hunk 3]
```diff
@@ -84,7 +84,7 @@ def admin_register(self, user, pool, admin_reason, feedback=''):
         :param feedback: Feedback to organizers
         :return: The registration
         """
-        if self.pools.filter(id=pool.id).exists():
+        if not pool or self.pools.filter(id=pool.id).exists():
```
Comment (hanswilw) [inline lego/apps/events/models.py]: What do you think about doing 
```
if pool and not self.pools.filter(id=pool.id).exists():
    raise NoSuchPool()
with transaction.atomic():
    ...
```?
Nitpicking, but I like avoid indenting the logic like this

**Comment by eirsyl** [comment]
@hanswilw

**Pull request merged**